### PR TITLE
CVE-2016-1619, CVE-2015-6755

### PR DESCRIPTION
**Pull request by nxc5929**
Updated CVE files for:
CVE-2016-1619 
CVE-2015-6755

Nathan Connor

### CODE REVIEW

File: cves/CVE-2015-6775.yml [hunk 1]
```diff
@@ -32,7 +32,7 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: 
+description: A static method which doesn't use signitures could allow a remote attacker to cause denial of service or possibly access secure information by creating a "fake" object and paasing it to this static method.
```
Comment (MEF4232) [inline cves/CVE-2015-6775.yml]: Please make this more than one line to make it more readable

Comment (MEF4232) [reply cves/CVE-2015-6775.yml]: *signatures
*passing

File: cves/CVE-2015-6775.yml [hunk 2]
```diff
@@ -91,20 +96,20 @@ discovered:
 
     If there is no evidence as to how this vulnerability was found, then you may
     leave this part blank.
-  answer: 
-  date: 
-  automated: 
-  google: 
-  contest: 
+  answer: The vulnerability was discover by Atte who used ClusterFuzz to discover the vulnerability.
```
Comment (MEF4232) [inline cves/CVE-2015-6775.yml]: *discovered

Comment (MEF4232) [reply cves/CVE-2015-6775.yml]: Who is Atte?

File: cves/CVE-2015-6775.yml [hunk 3]
```diff
@@ -113,10 +118,16 @@ interesting_commits:
...
     emerging themes?
   commits:
-  - commit: 
-    note: 
-  - commit: 
-    note: 
+  - commit: 1af240cc45480520b447be767686e73a29c48f9e
+    note: Refactored code and removed several dead functions. I'm surprised that a security review was never done during this refactoring to catch this vulnerability.
+  - commit: 506df426d5d64d68e9dc27ffebcf56f6c6a1bccf
+    note: Noted that it was hard to tell which functions come from V8 and another file. It was clear that the files in question where confusing. This may have lead to people not wanting to review the code.
```
Comment (MEF4232) [inline cves/CVE-2015-6775.yml]: *were
*led

File: cves/CVE-2015-6775.yml [hunk 4]
```diff
@@ -158,9 +169,9 @@ lessons:
   native_wrappers:
     applies: 
     note: 
-  distrust_input:
-    applies: 
-    note: 
+  distrust_input: 
+    applies: true
+    note: This vulnerability allowed for any object to be passed into the function without a signiture authorivation
```
Comment (MEF4232) [inline cves/CVE-2015-6775.yml]: *signature 
*authorization

File: cves/CVE-2015-6775.yml [hunk 5]
```diff
@@ -191,4 +202,4 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer: 
+  answer: The Chromuim team's inital commit contained a bug. This snipit of code went through multiple code reviews and tests yet still manages to exist in the system. The code was noted as confusing several times in the commits and yet nothing was not to simplify the code or seperate it out into multiple file for easier readability. Complex code tends to hide vulnerabilities such as this one and needs to be carefully reviewed before sent to production.
```
Comment (MEF4232) [inline cves/CVE-2015-6775.yml]: *snippet
*separate
*multiple files

I don't understand the section "yet nothing was not to simplify the code". Did you mean "yet nothing was done to simplify the code"?

File: cves/CVE-2016-1619.yml [hunk 6]
```diff
@@ -32,7 +32,7 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: 
+description: A PDF file containing a malformed JP2 image can causes an out-of-bounds read in the openjpeg library. Remote attackers are able to cause a denial of servie (out-of-bounds read) via a created PDF document.
```
Comment (MEF4232) [inline cves/CVE-2016-1619.yml]: *can cause

Comment (MEF4232) [reply cves/CVE-2016-1619.yml]: How is the out-of-bounds caused by the PDF document?

File: cves/CVE-2016-1619.yml [hunk 7]
```diff
@@ -76,9 +79,9 @@ unit_tested:
 
     For the "fix" answer below, check if the fix for the vulnerability involves
     adding or improving an automated test to ensure this doesn't happen again.
-  answer: 
-  code: 
-  fix: 
+  answer: The original vulnerability was never texted for in unit testing. The fix added a unit test which tests opening a pdf file contanining a JP2 known to cause the error.
```
Comment (MEF4232) [inline cves/CVE-2016-1619.yml]: *tested

Comment (MEF4232) [reply cves/CVE-2016-1619.yml]: What is the expected result from the new unit test? Like, if the system tries to open a pdf file contanining a JP2, how does the system react? What errors does it throw? Does it inform the user of the issue?

File: cves/CVE-2016-1619.yml [hunk 8]
```diff
@@ -180,8 +181,8 @@ lessons:
     applies: 
     note: 
   complex_inputs:
-    applies: 
-    note: 
+    applies: true
+    note: The codex used to render generate and render PDF's are complex. Therefore errors are likely to occur.
```
Comment (MEF4232) [inline cves/CVE-2016-1619.yml]: What does this mean "used to render generate"? What is "generate"?

File: cves/CVE-2016-1619.yml [hunk 9]
```diff
@@ -194,4 +195,4 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer: 
+  answer: Codex code is very complex and is generally created and then left alone. Software engineers don't want to have to continuouly look how a PDF is rendered and prefer to just assume it works as intended and leave it alone. This may be proplamatic for changing systems. This vulnerability could have lead to a DOS attack by crashing the server. Software engineers must be werry of what libraries they are using and protect there system from outside vulnerabilities.
```
Comment (MEF4232) [inline cves/CVE-2016-1619.yml]: *continuously
*problematic
*could have led
*wary 
*protect their system

File: cves/CVE-2015-6775.yml [hunk 10]
```diff
@@ -45,14 +45,19 @@ bounty:
...
-repo: 
+repo: https://pdfium.googlesource.com/pdfium
 fixes_vcc_instructions: |
   Please put the commit hash in "commit" below (see my example in
   CVE-2011-3092.yml). Fixes and VCCs follow the same format.
-fixes: []
-vccs: []
+fixes:
+- :commit: 158e335717efba9dce3aa6f6d1e31ed884e1f59e
+  :note: Added signitures
```
Comment (Jeng22) [inline cves/CVE-2015-6775.yml]: *signatures again

File: cves/CVE-2015-6775.yml [hunk 11]
```diff
@@ -113,10 +118,16 @@ interesting_commits:
...
+  - commit: 1af240cc45480520b447be767686e73a29c48f9e
+    note: Refactored code and removed several dead functions. I'm surprised that a security review was never done during this refactoring to catch this vulnerability.
+  - commit: 506df426d5d64d68e9dc27ffebcf56f6c6a1bccf
+    note: Noted that it was hard to tell which functions come from V8 and another file. It was clear that the files in question where confusing. This may have lead to people not wanting to review the code.
+  - commit: 09ed30750282bf56a92d0e646ab22c64bea81a36
+    note: File was refactored again yet error still lives on. 
+  - commit: 6df59849472958e7de96da6d9fc7b223b7c1f1c3
+    note: Was noted that it was hard to keep the mapping between v8 and fx abstractions.
+  - commit: b048f791a15f2da781a01eba5b09eb9d389f9c11
+    note: clang-format all code. Because Google decided tabs are better (=
```
Comment (Jeng22) [inline cves/CVE-2015-6775.yml]: Interesting, what is clang-format?

File: cves/CVE-2015-6775.yml (hunk 1 above)
Comment (Jeng22) [reply cves/CVE-2015-6775.yml]: Good Description

File: cves/CVE-2015-6775.yml [hunk 12]
```diff
@@ -125,7 +136,7 @@ major_events:
 
     The event doesn't need to be directly related to this vulnerability, rather,
     we want to capture what the development team was dealing with at the time.
-  answer: 
+  answer: Didn't find any events
```
Comment (Jeng22) [inline cves/CVE-2015-6775.yml]: You mention previously that this went through several reviews and remained unfixed. Wouldn't the actual fix be a significant event to mention?

File: cves/CVE-2015-6775.yml (hunk 1 above)
Comment (Jeng22) [reply cves/CVE-2015-6775.yml]: Agreed on using multiple lines.
Use "|" like in other fields to do this

File: cves/CVE-2016-1619.yml [hunk 13]
```diff
@@ -94,20 +97,20 @@ discovered:
...
-  answer: 
-  date: 
-  automated: 
-  google: 
-  contest: 
+  answer: Keve Nagy discovered an out-of-bounds-read issue in the pdfium library. Keve included a PDF file containing a JP2 image which caused the error to occur.
+  date: 11/17/15
+  automated: no
+  google: no
+  contest: no
```
Comment (Jeng22) [inline cves/CVE-2016-1619.yml]: Was Keve Nagy an outside reviewer if not from google or contest? If so is there a group they are associated with?

File: cves/CVE-2016-1619.yml (hunk 9 above)
Comment (Jeng22) [reply cves/CVE-2016-1619.yml]: Not everyone may know what a DOS attack is, should write out the acronym like the initial description

File: cves/CVE-2015-6775.yml (hunk 11 above)
Comment (MEF4232) [reply cves/CVE-2015-6775.yml]: I agree, tabs are better :)

File: cves/CVE-2016-1619.yml [hunk 14]
```diff
@@ -94,20 +97,20 @@ discovered:
 
     If there is no evidence as to how this vulnerability was found, then you may
     leave this part blank.
-  answer: 
-  date: 
-  automated: 
-  google: 
-  contest: 
+  answer: Keve Nagy discovered an out-of-bounds-read issue in the pdfium library. Keve included a PDF file containing a JP2 image which caused the error to occur.
```
Comment (MEF4232) [inline cves/CVE-2016-1619.yml]: What is the pdfium library?

**Pull request merged**